js/src/admin/index.js

```diff
@@ -1,53 +1,63 @@
-                    {app.translator.trans('fof-best-answer.admin.settings.schedule_stop_overnight')}
-                </BooleanItem>,
-                <BooleanItem setting={e} name="fof-best-answer.store_log_output">
-                    {app.translator.trans('fof-best-answer.admin.settings.schedule_log_output')}
-                </BooleanItem>,
-            ],
-        });
-
-    extend(PermissionGrid.prototype, 'replyItems', function(items) {
-        items.add('selectBestAnswer', {
+app.initializers.add('fof-best-answer', () => {
+    app.extensionData.for('fof-best-answer')
+        .registerSetting({
+            setting: 'fof-best-answer.allow_select_own_post',
+            type: 'boolean',
+            label: app.translator.trans('fof-best-answer.admin.settings.allow_select_own_post')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.use_alternative_ui',
+            type: 'boolean',
+            label: app.translator.trans('fof-best-answer.admin.settings.use_alt_ui')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.select_best_answer_reminder_days',
+            type: 'number',
+            label: app.translator.trans('fof-best-answer.admin.settings.select_best_answer_reminder_days')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.remind_tag_ids',
+            type: 'string',
+            label: app.translator.trans('fof-best-answer.admin.settings.remind_tag_ids')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.schedule_on_one_server',
+            type: 'boolean',
+            label: app.translator.trans('fof-best-answer.admin.settings.schedule_on_one_server')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.stop_overnight',
+            type: 'boolean',
+            label: app.translator.trans('fof-best-answer.admin.settings.schedule_stop_overnight')
+        })
+        .registerSetting({
+            setting: 'fof-best-answer.store_log_output',
+            type: 'boolean',
+            label: app.translator.trans('fof-best-answer.admin.settings.schedule_log_output')
+        })
+        .registerPermission({
             icon: 'far fa-comment',
             label: app.translator.trans('fof-best-answer.admin.permissions.best_answer'),
-            permission: 'discussion.selectBestAnswerOwnDiscussion',
-        });
-
-        items.add('selectBestAnswerNotAuthor', {
+            permission: 'discussion.selectBestAnswerOwnDiscussion'
+        }, 'reply')
+        .registerPermission({
             icon: 'far fa-comment',
             label: app.translator.trans('fof-best-answer.admin.permissions.best_answer_not_own_discussion'),
-            permission: 'discussion.selectBestAnswerNotOwnDiscussion',
-        });
-    });
+            permission: 'discussion.selectBestAnswerNotOwnDiscussion'
+        }, 'reply')
+
+    // app.extensionSettings['fof-best-answer'] = () =>
```

Comment:
This should be good to eliminate if it works locally

Reply:
Ooh, I left this comment to remind me to ask you.... `min`, `max` and `placeholder`, I couldn't see how to implement them using the new `registerSetting`?